### ISSUE TITLE
Consider moving to Show-able functions once QuickCheck releases >2.15.0

### ISSUE THREAD

**Issue by nwf** [CTSRD-CHERI/TestRIG#37]
QuickCheck has a notion of `Show`-able functions in [Test.QuickCheck.Function](https://hackage.haskell.org/package/QuickCheck-2.15.0.1/docs/Test-QuickCheck-Function.html).  After the 2.15.0 release, the [requisite constructor was exported](https://github.com/nick8325/quickcheck/commit/3f999ef44ec9943f59ab4987bdbc031f74357160), making these more useful as general-purpose objects.  Consider moving the currently higher-order (that is, `->`-bearing) `MetaInfo` ctors, https://github.com/lowRISC/QuickCheckVEngine/blob/ab924b4ac037b45fd182f11d2c1a7e079b1dcd0b/src/QuickCheckVEngine/TestTypes.hs#L66-L67 over to these once that's possible?

**Comment by PeterRugg** [comment]
Sounds good: I imagine that would tidy up `MetaAssertLastVal` which is already saved and restored in the traces IIRC. I imagine we'd need to introduce some constraints to `MetaAssertCompound`: presumably the `Show` can't cope with arbitrarily complex functions over sequences of unbounded length?

**Comment by nwf** [comment]
Yeah, `MetaAssertCompound` could be tricky.  How would you feel about imposing the requirement that its `->`s be `static`?
